In [177]:
r_earth = 6370 * (10 ** 3)

def ice_ball_dictionary(earth_rads, iron_part, sio2_part):
    ice_part = 1 - iron_part - sio2_part

    radius = earth_rads * r_earth
    
    iron_radius = iron_part * radius
    sio2_radius = sio2_part * radius
    ice_radius = ice_part * radius

    key1 = int(iron_radius)
    key2 = int(iron_radius) + int(sio2_radius)
    key3 = int(radius)
    
    ice_ball_dict = {key1: [7678], key2: [3000], key3: [930]}
#key3 is a dictionary that maps to three dictionaries corresponding to iceIH, ice6, ice7
#key2
#key1 for Fe0.93Si0.07
    
    return(ice_ball_dict)

#for r in range(0.1, 4, 0.1):
    #for i in range(0.1, 0.9, 0.1):
        #pass

ice_ball_dictionary(1, 0.4, 0.4)

{2548000: [7678], 5096000: [2648], 6370000: [930]}

ice_ball_dict = {
    10: {9999999999999999999: [7678, 136.2 * (10 ** 9), 5.97, 'vinet']}, 

    30: {
        23 * (10 ** 9): {'sio2': {2.5 * (10 ** 9): [2648, 37.4 * (10 ** 9), 6.2, 'bm3'],
                                  8 * (10 ** 9): [2921, 96 * (10 ** 9), 8.4, 'bm3'], 
                                  23 * (10 ** 10): [4290, 309.9 * (10 ** 9), 4.59, 'bm3']},
                     'mg2sio4': {14 * (10 ** 9): [3221, 125 * (10 ** 9), 4.0, 'bm3'],
                                  18 * (10 ** 9): [3491, 160  * (10 ** 9), 4, 'bm3'],
                                  23 * (10 ** 10): [3548, 182.0 * (10 ** 9), 4.2, 'bm3']}
                     },
                     
                    120 * (10 ** 9): [4101, 256 * (10 ** 9), 4, 'bm3'],
                    99999999999999999: [4058, 221 * (10 ** 9), 4.2, 'vinet']
                 }, 
                20: {0:[930, 9.85 * (10 ** 9), 6.6, 'murnaghan'],
                     1 * (10 ** 9):[1271, 14.05 * (10 ** 9), 4, 'bm3'],
                     2.1 * (10 ** 9):[1456,14.9 * (10 ** 9), 5.4, 'bm3']}
                }

for cutoff in list(ice_ball_dict.keys()):

    for pressure_bound in list(ice_ball_dict[cutoff].keys()):
        print('pressure_bound', pressure_bound)
        if type(ice_ball_dict[cutoff][pressure_bound]) == dict:
            #print(list(ice_ball_dict[cutoff][pressure_bound].values())[0])
            for sio2_secondary_pressure_bound in list(ice_ball_dict[cutoff][pressure_bound].values())[0]:
                for mg2sio4_secondary_pressure_bound in list(ice_ball_dict[cutoff][pressure_bound].values())[1]:
                
                    print('sio2_secondary_pressure_bound', sio2_secondary_pressure_bound)
                    print('mg2sio4_secondary_pressure_bound', mg2sio4_secondary_pressure_bound)
                    newdick = (ice_ball_dict[cutoff][pressure_bound])
                    print(newdick['sio2'][sio2_secondary_pressure_bound][0])
                    print(newdick['mg2sio4'][mg2sio4_secondary_pressure_bound][0])
            #print(list(ice_ball_dict[cutoff][pressure_bound].values())[1])
            print('next')
        else:
            print(ice_ball_dict[cutoff][pressure_bound][0])
